### ISSUE TITLE
Possible bug in KL divergence

### ISSUE THREAD

**Issue by cryptexcode** [pytorch/pytorch#5282]
I think there is a bug in the implementation of KL divergence (torch.nn.KLDivLoss). After getting wrong results, I checked the doc and found the following equation.
 loss(x, target) = 1/n \sum(target_i * (log(target_i) - x_i))

But it should be like the following
loss(x, target) = 1/n \sum(target_i * (log(target_i) - log(x_i)))

- Pytorch version: '0.4.0a0+ff9d1ae'

**Comment by soumith** [comment]
the documentation of KLDivLoss specifies that the `input` already has to be a set of log-probabilities, i.e. probably the output of a LogSoftmax or Sigmoid layer...

> As with NLLLoss, the input given is expected to contain log-probabilities, however unlike ClassNLLLoss, input is not restricted to a 2D Tensor, because the criterion is applied element-wise.
